src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb

```diff
@@ -1075,9 +1075,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                                                              builder.Add(New VariableDeclaration(declaration.LocalSymbol, declaration.InitializerOpt, declaration.Syntax))
                                                                          ElseIf base.Kind = BoundKind.AsNewLocalDeclarations Then
                                                                              Dim asNewDeclarations = DirectCast(base, BoundAsNewLocalDeclarations)
+                                                                             Dim localSymbolsBuilder = ArrayBuilder(Of ILocalSymbol).GetInstance()
                                                                              For Each asNewDeclaration In asNewDeclarations.LocalDeclarations
-                                                                                 builder.Add(New VariableDeclaration(asNewDeclaration.LocalSymbol, asNewDeclarations.Initializer, asNewDeclaration.Syntax))
+                                                                                 localSymbolsBuilder.Add(asNewDeclaration.LocalSymbol)
                                                                              Next
+                                                                             builder.Add(New VariableDeclaration(localSymbolsBuilder.ToImmutableAndFree(), asNewDeclarations.Initializer, asNewDeclarations.Syntax))
```

Comment:
Consider using `asNewDeclarations.SelectAsArray(Function(d) d.LocalSymbol)` instead of `For Each`.